todo-app/src/App.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useEffect } from "react";
+import './App.css'
+import ToDo from "./components/ToDo";
+import TodoFilter from "./components/TodoFilter";
+
+const App = () => {
+    const [input, setInput] = useState('');
+    const [todos, setTodos] = useState([]);
+    const [criteria, setCriteria] = useState('All')
+
+    useEffect(() => {
+        const storedTodos = JSON.parse(localStorage.getItem('todos')) || [];
+        setTodos(storedTodos);
```

Comment:
it's optional but would be good to add a condition that 'storedTodos' have at least one todo in array

and if the condition is true, then you can do 'setTodos'

this way, you won't call a setter if no todos are in localStorage which means you'll avoid unnecessary rerender (cause setter triggers a rerender)

Reply:
No, move `JSON.parse(localStorage.getItem('todos')) || []` inside of todos `useState` directly. Set it as an initial value